src/transformers/processing_utils.py

```diff
@@ -594,6 +631,14 @@ def get_processor_dict(
                     f" directory containing a {PROCESSOR_NAME} file"
                 )
 
+        # Add chat template as kwarg before returning because most models don't have processor config
+        chat_template = None
+        if resolved_chat_template_file is not None:
+            with open(resolved_chat_template_file, "r", encoding="utf-8") as reader:
+                text = reader.read()
+            chat_template = json.loads(text)["chat_template"]
```

Comment:
Is there a reason for doing this in two steps i.e. getting `text` then `json.loads` instead of `json.load` in the reader context directly?

Reply:
ah, will simplify it. I was copying from processor

Reply:
Can't be loaded as `json` because it was saved as `TextIOWrapper`. Actually I don't know why we save it this way, was copying from processors. Maybe it has something to do with safe-saving 🤔

Reply:
huh, interesting. Well, good to know :) Thanks for investigating